### PR TITLE
Avoid hasChildren deprecation

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Fixed
- Avoid deprecation for the `hasChildren` method.
```

### CODE REVIEW

**Review by VincentLanglet — DISMISSED** [review]

**Comment by VincentLanglet** [comment]
Since we're rely on the fact we're iterating on Blocks, I removed the `@phpstan-template T` on 5.x
see https://github.com/sonata-project/SonataBlockBundle/blob/5.x/src/Util/RecursiveBlockIterator.php#L22

**Comment by jordisala1991** [comment]
sad I will need to open the code to fix this... tomorrow then.

**Review by VincentLanglet — DISMISSED** [review]

File: src/Model/BaseBlock.php [hunk 1]
```diff
@@ -199,7 +199,7 @@ public function addChildren(BlockInterface $children): void
         $children->setParent($this);
     }
 
-    public function getChildren(): iterable
+    public function getChildren()
```
Comment (jordisala1991) [inline src/Model/BaseBlock.php:202]: Had to change this to fix phpstan

File: src/Model/BlockInterface.php [hunk 2]
```diff
@@ -92,9 +94,9 @@ public function addChildren(self $children): void;
     /**
      * NEXT_MAJOR: Restrict typehint to Collection.
      *
-     * @return iterable<BlockInterface> $children
+     * @return Collection<int, BlockInterface>|array<BlockInterface> $children
      */
-    public function getChildren(): iterable;
+    public function getChildren();
```
Comment (jordisala1991) [inline src/Model/BlockInterface.php:99]: And this.

Comment (VincentLanglet) [reply src/Model/BlockInterface.php:99]: I think you can keep the return type.
```
public function getChildren(): iterable;
```

**Review by VincentLanglet — APPROVED** [review]

File: src/Model/BlockInterface.php (hunk 2 above)
Comment (jordisala1991) [reply src/Model/BlockInterface.php:99]: It breaks phpstan.

**Pull request merged**